### PR TITLE
WT-7410 Split session flags into flags and lock_flags

### PR DESCRIPTION
**Pull request by kommiharibabu**
Splitting the existing session flags into two flags can
accommodate more future flags

### CODE REVIEW

**Comment by kommiharibabu** [comment]
The stress test failure is a known issue of WT-7267.

**Review by sueloverso — APPROVED** [review]
LGTM with one minor comment.

File: src/log/log_slot.c [hunk 1]
```diff
@@ -526,7 +526,7 @@ __wt_log_slot_join(WT_SESSION_IMPL *session, uint64_t mysize, uint32_t flags, WT
     log = conn->log;
     time_start = time_stop = 0;
 
-    WT_ASSERT(session, !F_ISSET(session, WT_SESSION_LOCKED_SLOT));
+    WT_ASSERT(session, !FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_SLOT));
```
Comment (quchenhao) [inline src/log/log_slot.c:529]: Can we create a macro to check the lock flags? I feel it is accessed a lot.

**Review by quchenhao — APPROVED** [review]
lgtm

File: src/conn/conn_log.c [hunk 2]
```diff
@@ -1028,7 +1028,7 @@ int
 __wt_logmgr_open(WT_SESSION_IMPL *session)
 {
     WT_CONNECTION_IMPL *conn;
-    uint32_t session_flags;
+    uint32_t session_flags, session_lock_flags;
```
[the remark this answers is not in the document]
Comment (kommiharibabu) [reply src/conn/conn_log.c:1031]: removed the local variable.

File: src/log/log_slot.c (hunk 1 above)
Comment (kommiharibabu) [reply src/log/log_slot.c:529]: Expect for `flags` in the code we are currently referring to the `*_flags` directly wherever they are accessed. Adding a macro also should be specific to the session as the `lock_flags` is specific to the session. I don't find that it improves the code much.

**Pull request merged**